pkg/utils/wildcard.go

```diff
@@ -0,0 +1,24 @@
+package utils
+
+import (
+	"path/filepath"
+)
+
+// MatchWildcard checks if a string matches a wildcard pattern.
+// The pattern can include simple file-style glob patterns:
+// - '*' matches any sequence of non-separator characters
+// - '?' matches any single non-separator character
+// - '[abc]' matches any character within the brackets
+// - '[a-z]' matches any character in the range
+func MatchWildcard(pattern, str string) (bool, error) {
```

Comment:
Pretty sure we cannot use filepath.Match. Everywhere we support wildcards, we are really supporting globs with `**` support.